cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -2606,15 +2611,25 @@ public void showTopicsDialog(Activity dialogActivity, TopicsDialogListener topic
 
         checkboxLayout.addView(scrollView);
 
-        AlertDialog.Builder alertBuilder =
-            getTopicAlertBuilder(dialogActivity, topicsDialogListener, themeResId, channelConfig, channelTopics,
-                hasDeSelectAllInitial, checkboxLayout, unsubscribeCheckbox, selectedTopics);
-        AlertDialog alert = alertBuilder.create();
-        alert.setOnShowListener(dialog -> {
-          Logger.d(LOG_TAG, "showTopicsDialog activity: " + dialogActivity.getClass().getCanonicalName());
-          showingTopicsDialog = true;
-        });
-        alert.show();
+        try {
+          AlertDialog.Builder alertBuilder =
+                  getTopicAlertBuilder(dialogActivity, topicsDialogListener, themeResId, channelConfig, channelTopics,
+                          hasDeSelectAllInitial, checkboxLayout, unsubscribeCheckbox, selectedTopics);
+          AlertDialog alert = alertBuilder.create();
+          alert.setOnShowListener(dialog -> {
+            Logger.d(LOG_TAG, "showTopicsDialog activity: " + dialogActivity.getClass().getCanonicalName());
+            showingTopicsDialog = true;
+          });
+          Logger.d("Topic Dialog", "Test 1");
+          if (alert != null) {
+            alert.show();
+          }
+        } catch (Exception e) {
+          Logger.d(LOG_TAG, e.getLocalizedMessage());
+          if (!isRecursiveCall) {
```

Comment:
do we need to do the whole try catch if isRecursiveCall? i dont think so right? so this if can be around it

Reply:
@cpKevinrodriguesnobre Sorry. I didn't get it. Can you explain it more?

Reply:
![image](https://github.com/cleverpush/cleverpush-android-sdk/assets/99720107/d72cb5c4-75db-4ed6-b52a-3a35a75cb981)
does this help?

Reply:
No. It won't work. When it goes to catch and showTopicsDialog at that time isRecursiveCall flag will be true so it won't go inside that if and the dialog will not gonna be open. 

We do need not to call again showTopicsDialog method once it's true. that's why put it on catch block.